### PR TITLE
Custom button set reorder API

### PR DESCRIPTION
**Pull request by akhilkr128**
Adding Api for [task](https://github.com/ManageIQ/manageiq-ui-classic/issues/7901)

Dependent PR:
- [x] https://github.com/ManageIQ/manageiq/pull/21814

## Successful API call

Screenshot 2022-04-06 at 10 36 18 AM

## Call returns error

Screenshot 2022-04-06 at 10 36 02 AM

### CODE REVIEW

**Comment by akhilkr128** [comment]
@miq-bot cross-repo-test https://github.com/ManageIQ/manageiq/pull/21814

**Comment by akhilkr128** [comment]
@miq-bot cross-repo-test https://github.com/ManageIQ/manageiq/pull/21814

File: config/api.yml [hunk 1]
```diff
@@ -1210,6 +1210,8 @@
         :identifier: ab_group_edit
       - :name: delete
         :identifier: ab_group_delete
+      - :name: order
```
Comment (Fryguy) [inline config/api.yml:1213]: I'm thinking the action name should be `reorder`.  To me, `order`, sounds like you're going to purchase something.

File: app/controllers/api/custom_button_sets_controller.rb [hunk 2]
```diff
@@ -7,5 +7,12 @@ def create_resource(type, id, data)
     def edit_resource(type, id, data)
       super(type, id, data.deep_symbolize_keys)
     end
+
+    def order_resource(_, _, data)
+      CustomButtonSet.update_group_index(data["ids"])
```
Comment (Fryguy) [inline app/controllers/api/custom_button_sets_controller.rb:12]: I think we need an RBAC check here, or does that happen automatically?  cc @kbrock

Comment (Fryguy) [reply app/controllers/api/custom_button_sets_controller.rb:12]: @akhilkr128 Do you know what RBAC was used in the old UI?

File: app/controllers/api/custom_button_sets_controller.rb [hunk 3]
```diff
@@ -7,5 +7,12 @@ def create_resource(type, id, data)
     def edit_resource(type, id, data)
       super(type, id, data.deep_symbolize_keys)
     end
+
+    def order_resource(_, _, data)
```
Comment (Fryguy) [inline app/controllers/api/custom_button_sets_controller.rb:11]: @akhilkr128 As mentioned elsewhere, I'm thinking this should be `reorder`.  I'm concerned the term `order` might get confused with things like ordering services.

File: config/api.yml (hunk 1 above)
Comment (akhilkr128) [reply config/api.yml:1213]: @Fryguy Do you think `sort` will be a better alternative?

File: app/controllers/api/custom_button_sets_controller.rb (hunk 2 above)
Comment (akhilkr128) [reply app/controllers/api/custom_button_sets_controller.rb:12]: We need an admin role, correct me if I am wrong. 
see the [yml](https://github.com/ManageIQ/manageiq/blob/619680b323bc4305e23508cb40228225fdc5b85e/db/fixtures/miq_product_features.yml#L2583)

**Comment by akhilkr128** [comment]
@miq-bot cross-repo-test https://github.com/ManageIQ/manageiq/pull/21814

File: app/controllers/api/custom_button_sets_controller.rb [hunk 4]
```diff
@@ -7,5 +7,12 @@ def create_resource(type, id, data)
     def edit_resource(type, id, data)
       super(type, id, data.deep_symbolize_keys)
     end
+
+    def reorder_resource(_, _, data)
+      CustomButtonSet.reorder_group_index(data["ids"])
+      action_result(true, "Button Group Reorder saved")
+    rescue => error
+      action_result(false, error.to_s)
+    end
```
Comment (kbrock) [inline app/controllers/api/custom_button_sets_controller.rb:16]: Probably not necessary but I'd feel more comfortable using the `api_action` pattern

It does little things different error reports if groups of calls are bundled together.

```suggestion
    def reorder_resource(type, id, data)
      api_action(type, id) do
        CustomButtonSet.reorder_group_index(data["ids"])
        action_result(true, "Button Group Reorder saved")
      end
    end
```

**Review by kbrock — COMMENTED** [review]
this looks good.
I merged the core change so this should be 🍏 now.
just the one change and then we can merge

**Comment by akhilkr128** [comment]
@miq-bot cross-repo-test https://github.com/ManageIQ/manageiq/pull/21814

**Pull request merged**